cmd/network-tracer/tracer.go

```diff
@@ -61,9 +61,19 @@ func CreateNetworkTracer(cfg *config.AgentConfig) (*NetworkTracer, error) {
 
 // Run makes available the HTTP endpoint for network collection
 func (nt *NetworkTracer) Run() {
-	http.HandleFunc("/status", func(w http.ResponseWriter, req *http.Request) {})
+	httpMux := http.DefaultServeMux
 
-	http.HandleFunc("/connections", func(w http.ResponseWriter, req *http.Request) {
+	// If profiling is disabled, then we should overwrite handlers for the pprof endpoints
+	// that were registered in init():
+	// https://github.com/golang/go/blob/5bd88b0/src/net/http/pprof/pprof.go#L72-L78
+	// We can only do this by creating a new HTTP Mux that does not have these endpoints handled
+	if !nt.cfg.EnableDebugProfiling {
+		httpMux = http.NewServeMux()
```

Comment:
We were getting `panic: http: multiple registrations for /debug/pprof/` with the previous method of disabling the pprof endpoints. This method works and has been tested.